Bump Open Telemetry packages

stevenbuccini opened the pull request:
The previous versions of Open Telemetry raise deprecation warnings. Bump the packages to address these errors. Resolves #11069.

stevenbuccini committed:
Bump Open Telemetry packages

The previous versions of Open Telemetry raise deprecation warnings. Bump the packages to address these errors.

stevenbuccini commented:
Hey @yosefAlsuhaibani could you take a look at this since you were the one who originally debugged the issue? Thanks!

dijkstracula commented:
Hi @stevenbuccini , it looks like you have a bunch of CI failures: does local build and testing in a fresh venv pass for you locally?

1) The test suite fails because of a new missing dependency: I wonder if the otel version bump tacetly removed the pip rule to install `setuptools` (which I believe supplants `distutils` nowadays).
2) [tests / test semgrep-cli (3.9) (pull_request)](https://github.com/semgrep/semgrep/actions/runs/17214473190/job/48859485702?pr=11180) indicates that some output snapshots have changed.  If those are indeed valid changes they'll have to be incorporated into the PR (via `--snapshot-update`).

Cheers,
Nathan

stevenbuccini committed:
Migrate pytest-freezegun

Bumping the OTel collector packages meant that setuptools was no longer being installed as a dependency. This broke pytest-freezegun, which required distutils. pytest-freezegun is no longer being maintained, so this commit migrates to pytest-freezer which is an actively-maintained drop-in replacement.

Additionally, we removed the associated type hinting packages. From pypi: "The freezegun package includes type annotations or type stubs since version 1.2.1. Please uninstall the types-freezegun package if you use this or a newer version." Inspecting the Pipfile.lock shows that our current verison of freezegun is 1.5.5 so we can safely remove this package.

stevenbuccini commented:
> 1. The test suite fails because of a new missing dependency: I wonder if the otel version bump tacetly removed the pip rule to install `setuptools` (which I believe supplants `distutils` nowadays).

Good catch, looks like [that's the case](https://github.com/semgrep/semgrep/pull/11180/files#diff-afedf751e67c1e430dd5614c726862ac3e23d9d73840c178f26ec7269e71def8L491-L498). Is there a reason `setuptools` isn't explicitly listed in the pipfile?

Looks like offending package [is actually `pytest-freezegun`](https://github.com/ktosiek/pytest-freezegun/issues/35), which is deprecated. I will migrate the package to its successor and see if that fixes the issue.

stevenbuccini committed:
Update snapshots

Automatically generated by `pipenv run pytest --snapshot-update`

stevenbuccini commented:
Hey @dijkstracula, should be good to go now. Tests are all passing locally for me.

yosefAlsuhaibani commented:
Hi @stevenbuccini: this PR should be superseded by https://github.com/semgrep/semgrep/commit/96f4032b4439aa488772156a9363ee01b06a3cdb. Sorry this took as long as it did; for some reason upgrading the open-telemetry dep via `pipenv lock` upgraded a bunch of (possibly transitive) dependencies, which broke our pytest snapshots (as text formatting behavior was different).
